### ISSUE TITLE
Silence spam output from noisy tests

### ISSUE THREAD

**Issue by dcolligan** [ga4gh/server#253]
A recently added test is spewing out extraneous debug output.  This shouldn't happen -- silence this test!

```
$ nosetests -v tests/datadriven/
...
tests.datadriven.test_variants.VariantSetTest.testVariantSetMetadata:1kgPhase1 ... ok
[W::_vcf_parse_format] FORMAT 'GT' is not defined in the header, assuming Type=String
[W::_vcf_parse_format] FORMAT 'GT' is not defined in the header, assuming Type=String
[W::_vcf_parse_format] FORMAT 'GT' is not defined in the header, assuming Type=String
tests.datadriven.VariantSetTest.testProtocolElementValid:1kgPhase3 ... ok
...
```

One way to do this might be creating a `@SuppressOutput` decorator in `tests.utils` that redirects stdout and stderr to `/dev/null` for the running duration of the test and then restores the file handles at the end.

**Comment by dcolligan** [comment]
Alternatively, fix the data file (looks like one of the 1kg VCF files has a malformed header line)

**Comment by jeromekelleher** [comment]
These warnings are coming from htslib, which would be nice to suppress in general; is there a way to silence it through pysam?

**Comment by dcolligan** [comment]
The error looks like it comes from this line in htslib:

https://github.com/samtools/htslib/blob/f3e1602196bbf03f426dfb363a4841932a042194/vcf.c#L1522

It looks like it's writing it directly to stderr without going through any kind of logging subsystem.  So there are probably no hooks in pysam or htslib to redirect / silence.

**Comment by dcolligan** [comment]
@jeromekelleher do you know what line in our code is accessing the htslib code that is emitting the error?

**Comment by dcolligan** [comment]
Nevermind, seems to be `DataDrivenTest.__init__` where `self._gaObject` is initialized

**Comment by dcolligan** [comment]
Redirecting stdout and stderr doesn't seem to do it... this diff still emits the output...
```
     def __init__(self, setId, baseDir):
         self._setId = setId
         self._dataDir = os.path.join(baseDir, setId)
+        origStdout = sys.stdout
+        origStderr = sys.stderr
+        devnull = open(os.devnull, 'w')
+        sys.stdout = devnull
+        sys.stderr = devnull
         self._gaObject = self.getDataModelClass()(
             self._setId, self._dataDir)
+        sys.stdout = origStdout
+        sys.stderr = origStderr
```
Not sure what the deal is here.  Maybe stderr gets lots in the C/python binding?

**Comment by dcolligan** [comment]
Aha, looks like python's stderr object is a wrapper around something but C accesses the file descriptors directly.

http://stackoverflow.com/questions/5081657/how-do-i-prevent-a-c-shared-library-to-print-on-stdout-in-python

**Comment by dcolligan** [comment]
Fixed in #261
